### DISCUSSION TITLE
Need advice in modelling and structuing

### DISCUSSION THREAD

**Discussion by Mayank-84** [pointfreeco/swift-composable-architecture#2881] [Q&A] [answered]
Hi everyone, I am new to TCA and I am already loving it for the benefits it offers. I was trying to make a clone of Facebook social home feed. So I was thinking to structure it like this

- HomeFeedFeature (parent) - It will have IdentifiedArrayOf HomeFeedModel
- IndividualHomeFeedFeature(child) - This will manage the individual post, interactions like "like", comment, share, etc

But the problem I am facing is I don't know how to make array of child states within the parent and derive the corrsesponding store to pass to the child.  I saw in CounterFeature example mentioned in documentation, but that was only one child. But how do I deal with array of child states and actions and eventually derive a store to initialise the corresponding child.

Can you guys please help me in pointing to any TCA concept that can fix this elegantly? Or any sugesstions in terms of modelling. 

Thanks you.

**Comment by kamcma** [comment, marked as answer]
The way you do this is have the parent hold an `IdentifiedArray<ChildFeature.State>` in its state and have an action case `childAction(IdentifiedAction<ChildFeature.Action>)`. Then in your reducer body, there is [a `.forEach()` operator](https://pointfreeco.github.io/swift-composable-architecture/1.9.0/documentation/composablearchitecture/reducer/foreach(_:action:element:fileid:line:)-247po) you point at this state and action and the child reducer.

In the case studies in this repository, there are several examples having to do with lists. Have a look at those.

**Reply by Mayank-84** [reply]
Thank you.

**Reply by kamcma** [reply]
Oh, I should I have thought to recommend the demo Todo app as well, also in the examples folder. That's probably even better than the lists case studies for this particular topic.
